javaparser-core/src/main/java/com/github/javaparser/ast/visitor/DumpVisitor.java

```diff
@@ -526,7 +527,13 @@ public void visit(final ReferenceType n, final Object arg) {
 			}
 		}
 		n.getType().accept(this, arg);
-		printArrayBraces(n, arg);
+	}
+
+	@Override
+	public void visit(ArrayType n, Object arg) {
```

Comment:
Unfortunately, if the ArrayType wrapping order is fixed (see comment bellow in the grammar), this can't be handled this easily. This must be handled with a loop (or a collecting inner-visitor) unwrapping all inner ArrayTypes up to a (non-Array)Type.

For instance, for a type 
`ArrayType(annotations=@B, ArrayType(annotations=@C, PrimitiveType(annotations=@A, primitive=int)))`
should be printed as
`@A int @B [] @C []`